src/core/public/saved_objects/saved_objects_client.ts

```diff
@@ -351,11 +420,37 @@ export class SavedObjectsClient {
     return resolveUrl(API_BASE_URL, join(...path));
   }
 
-  private request({ method, path, query, body }: RequestParams) {
-    if (method === 'GET' && body) {
-      return Promise.reject(new Error('body not permitted for GET requests'));
-    }
-
-    return kfetch({ method, pathname: path, query, body: JSON.stringify(body) });
+  /**
+   * To ensure we don't break backwards compatibility, savedObjectsFetch keeps
+   * the old kfetch error format of `{res: {status: number}}` whereas `http.fetch`
+   * uses `{response: {status: number}}`
+   */
+  private savedObjectsFetch(path: string, { method, query, body }: HttpFetchOptions) {
+    return this.http.fetch(path, { method, query, body }).catch(err => {
+      const kfetchError = Object.assign(err, { res: err.response });
```

Comment:
btw should we deprecate `res` field on the result object?

Reply:
Yes good point. I think it would be good to have error helper functions like the Server SavedObjectsClient so that error handling isn't hardcoded to error shapes which we can't type because these are returned in an exception. I've created a task for it in https://github.com/elastic/kibana/issues/33587 I've also removed `err.response` so that we avoid confusion between the two.